source/material_model/visco_plastic.cc

```diff
@@ -303,8 +304,71 @@ namespace aspect
 
       if (this->get_parameters().enable_elasticity)
         {
+          // Fill the elastic outputs with the body force term for the RHS, the viscoelastic strain rate
+          // and the viscous dissipation.
           rheology->elastic_rheology.fill_elastic_outputs(in, average_elastic_shear_moduli, out);
+          // Fill the reaction terms that account for the rotation of the stresses.
           rheology->elastic_rheology.fill_reaction_outputs(in, average_elastic_shear_moduli, out);
+          // Fill the reaction_rates that apply the stress update of the previous
+          // timestep to the advected and rotated stress computed in the previous timestep ($\tau^{0}$)
+          // to obtain $\tau^{t}$.
+          // Only fill them if operator splitting is used for fields or when the particles track
+          // the visco-elastic stresses.
+          if (this->get_parameters().use_operator_splitting ||
+              ((this->get_parameters().mapped_particle_properties).count(this->introspection().compositional_index_for_name("ve_stress_xx"))))
```

Comment:
This condition could use an update to the compositional type machinery: `this->introspection().composition_type_exists(CompositionalFieldDescription::stress)`

Reply:
I think the check can be removed. If elasticity is used with the VE or VP material model, we already do this check in parse_parameters in elasticity.cc. I did add the check `in.requests_property(MaterialProperties::reaction_rates))` in fill_reaction_rates.